apps/settings/js/panels/homescreens_list/homescreens_list.js

```diff
@@ -0,0 +1,189 @@
+        name: manifest.name,
+        author: manifest.developer,
+        version: manifest.version,
+        description: manifest.description,
+        removable: app.removable,
+        app: app
+      });
+    },
+
+    /**
+     * Build list item.
+     *
+     * @param {Object} app   app object.
+     * @param {Number} index link index.
+     * @private
+     */
+    _listItemBuilder: function hl_listItemBuilder(app, index) {
+      var item = document.createElement('li');
+      item.className = 'pack-split';
+      item.dataset.appIndex = index;
+
+      var radioLabel = document.createElement('label');
+      radioLabel.className = 'pack-radio';
+      radioLabel.role = 'presentation';
+
+      var input = document.createElement('gaia-radio');
+      input.name = 'homescreen';
+      input.value = index;
+      input.checked = (app.manifestURL === this.manifestURL);
+      radioLabel.appendChild(input);
+
+      item.appendChild(radioLabel);
+
+      var nameLabel = document.createElement('label');
+      nameLabel.className = 'name';
+
+      var manifest =
+        new ManifestHelper(app.manifest || app.updateManifest);
+      var icon = document.createElement('img');
+      if (manifest.icons && Object.keys(manifest.icons).length) {
+        var key = Object.keys(manifest.icons)[0];
+        var iconURL = manifest.icons[key];
+        if (!(/^(http|https|data):/.test(iconURL))) {
+          iconURL = app.origin + '/' + iconURL;
+        }
+        icon.src = iconURL;
+      } else {
+        icon.src = '../style/images/default.png';
+      }
+      nameLabel.appendChild(icon);
+
+      var span = document.createElement('span');
+      span.textContent = manifest.name;
+      nameLabel.appendChild(span);
+
+      item.appendChild(nameLabel);
+
+      return item;
+    },
+
+    /**
+     * Build the home screens list
+     */
+    listBuilder: function hl_listBuilder() {
+      this.elements.homescreensList.innerHTML = '';
+      var listFragment = document.createDocumentFragment();
+      var item;
+      this._apps.forEach((app, index) => {
+        item = this._listItemBuilder(app, index);
+        listFragment.appendChild(item);
+      });
+      this.elements.homescreensList.appendChild(listFragment);
+    },
+
+    /**
+     * @private
+     */
+    _renderHomescreens: function hl_renderHomescreens() {
+      return AppsCache.apps().then(apps => {
+        this._apps = apps.filter(app => {
+          var manifest =
+            new ManifestHelper(app.manifest || app.updateManifest);
+          return manifest && manifest.role && manifest.role === 'homescreen';
+        });
+
+        this.listBuilder();
+      });
+    },
+
+    /**
+     * @private
+     */
+    _showMoreHomescreens: function hl_showMoreHomescreens() {
+      SettingsCache.getSettings(result => {
+        var version = result['deviceinfo.os'];
+        /* jshint nonew: false */
+        new window.MozActivity({
+          name: 'marketplace-category',
```

Comment:
The creation of this category is tracked in [Bug 1193284 - Add a category for replaceable home screens](https://bugzilla.mozilla.org/show_bug.cgi?id=1193284).